### DISCUSSION TITLE
Proper way to put attributes in generated properties when using sour generators

### DISCUSSION THREAD

**Discussion by cesarchefinho** [CommunityToolkit/dotnet#217] [Q&A] [answered]
What is the proper way to preserve attributes of observable properties when using source generators?

Example

```csharp
private string? firstName;

[Required]
[CustomValidatorBlaBla] 
public string? FirstName
{
    get => firstName;
    set
    {
        if (SetProperty(ref firstName, value))
        {
            OnPropertyChanged(nameof(FullName));
            GreetUserCommand.NotifyCanExecuteChanged();
        }
    }
}

private string? lastName;

public string? LastName
{
    get => lastName;
    set
    {
        if (SetProperty(ref lastName, value))
        {
            OnPropertyChanged(nameof(FullName));
            GreetUserCommand.NotifyCanExecuteChanged();
        }
    }
}

public string? FullName => $"{FirstName} {LastName}";
```

Whne using generators we code as bellow

```csharp
[ObservableProperty]
[AlsoNotifyChangeFor(nameof(FullName))]
[AlsoNotifyCanExecuteFor(nameof(GreetUserCommand))]
private string? firstName;

[ObservableProperty]
[AlsoNotifyChangeFor(nameof(FullName))]
[AlsoNotifyCanExecuteFor(nameof(GreetUserCommand))]
private string? lastName;

public string? FullName => $"{FirstName} {LastName}";
```

How I make Required and CustomValudatorBlaBla be applied to FirstName property generetaded?

**Comment by Sergio0694** [comment]
You can just use the attribute, and it'll be automatically ported over to the generated property 🙂
Note: this only applies to validation attributes.

**Reply by cesarchefinho** [reply]
But if I have other attributes in viewmodel from anottations namespace like DisplayNameAttribute or DisplayFormatAttribute? 

How can I propagate theses attributes using source generators to generated property's ?

**Reply by PetarSlavov** [reply]
I'm also facing the same problem. As a workaround I can inherit all my custom attributes from ValidationAttribute, but it looks like this only applies to properties not commands.

**Reply by michael-hawker** [reply]
Yeah, see issue #208

**Comment by cesarchefinho** [comment]
I resolve this issue using reflection and reading private fields attributes with same name of propretyes and uncaptalize the first letter of name.

**Comment by michael-hawker** [comment, marked as answer]
Tracked by issue #208
